src/faction_camp.cpp

```diff
@@ -1962,132 +2066,85 @@ bool basecamp::survey_return( npc &p )
-                    bay.set( x, y, t_dirt, f_plant_seed );
+    //bay_json is what the are should look like according to jsons
+    tinymap bay_json;
+    bay_json.generate( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, calendar::turn );
+    //bay is what the area actually looks like
+    tinymap bay;
+    bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+    tripoint mapmin = tripoint( 0, 0, omt_tgt.z );
+    tripoint mapmax = tripoint( 2 * SEEX - 1, 2 * SEEY - 1, omt_tgt.z );
+    for( const tripoint &pos : bay.points_in_rectangle( mapmin, mapmax ) ) {
+        //Needs to be plowed to match json
+        if( plow && is_dirtmound( pos, bay_json, bay ) && is_unplowed( pos, bay ) ) {
+            bay.ter_set( pos, t_dirtmound );
+        }
+        if( plant && is_dirtmound( pos, bay, bay ) && !seed_inv.empty() ) {
+            item *tmp_seed = seed_inv.back();
+            seed_inv.pop_back();
+            std::list<item> used_seed;
+            if( tmp_seed->count_by_charges() ) {
+                used_seed.push_back( *tmp_seed );
+                tmp_seed->charges -= 1;
+                if( tmp_seed->charges > 0 ) {
+                    seed_inv.push_back( tmp_seed );
                 }
             }
-            if( harvest && bay.furn( x, y ) == furn_str_id( "f_plant_harvest" ) &&
-                !bay.i_at( x, y ).empty() ) {
-                const item &seed = bay.i_at( x, y )[0];
-                if( seed.is_seed() && seed.typeId() != "fungal_seeds" &&
-                    seed.typeId() != "marloss_seed" ) {
-                    const itype &type = *seed.type;
-                    int skillLevel = comp->get_skill_level( skill_survival );
-                    ///\EFFECT_SURVIVAL increases number of plants harvested from a seed
-                    int plantCount = rng( skillLevel / 2, skillLevel );
-                    //this differs from
-                    if( plantCount >= 9 ) {
-                        plantCount = 9;
-                    } else if( plantCount <= 0 ) {
-                        plantCount = 1;
-                    }
-                    const int seedCount = std::max( 1l, rng( plantCount / 4, plantCount / 2 ) );
-                    for( auto &i : iexamine::get_harvest_items( type, plantCount, seedCount,
-                            true ) ) {
-                        g->m.add_item_or_charges( g->u.posx(), g->u.posy(), i );
-                    }
-                    bay.i_clear( x, y );
-                    bay.furn_set( x, y, f_null );
-                    bay.ter_set( x, y, t_dirt );
+            used_seed.front().set_age( 0 );
+            bay.add_item_or_charges( pos, used_seed.front() );
+            bay.set( pos, t_dirt, f_plant_seed );
+        }
+        if( harvest && bay.furn( pos ) == furn_str_id( "f_plant_harvest" ) && !bay.i_at( pos ).empty() ) {
+            const item &seed = bay.i_at( pos ).front();
+            const itype &type = *seed.type;
```

Comment:
This not needed here. Declare it right before where it's actually needed below. (Or just use `*seed.type` directly at the one place where it's referenced.)